components-java-sdk/src/main/java/io/dapr/components/aspects/InitializableWithProperties.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2023 The Dapr Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.dapr.components.aspects;
+
+import reactor.core.publisher.Mono;
+
+import java.util.Map;
+
+public interface InitializableWithProperties {
+  Mono<Void> init(Map<String, String> properties);
```

Comment:
The .NET SDK leaves the properties as part of a higher-level request argument, more closely mimicking the service contract.  At the time I wasn't sure whether it was conceivable (or not) that Dapr might, at some point, provide the component more than just the properties and didn't want to potentially end up with a string of optional arguments in order to maintain build compatibility.

I think both approaches have their advantages (i.e. this one is certainly simpler).  Once we have a complete set of APIs across the platforms, and perhaps have some customer feedback as to their shape, we should do a comparison of APIs to see where more consistency might make sense.

Reply:
Agreed. I just tried to not deviate too much from the .proto definition. Both because I thought it would be less controversial but because I think it would be worth exposing this to developers to see what they could come up with.